mm2src/mm2_bitcoin/chain/src/transaction.rs

```diff
@@ -528,6 +540,88 @@ where
         shielded_spends,
         shielded_outputs,
         zcash,
+        posv,
+        str_d_zeel,
+        tx_hash_algo: TxHashAlgo::DSHA256,
+    })
+}
+
+pub fn deserialize_tx_posv<T>(reader: &mut Reader<T>, tx_type: TxType) -> Result<Transaction, Error>
```

Comment:
Can you please remove this function and use the `deserialize_tx` function instead by applying the following changes
```diff
Index: mm2src/mm2_bitcoin/chain/src/transaction.rs
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/mm2src/mm2_bitcoin/chain/src/transaction.rs b/mm2src/mm2_bitcoin/chain/src/transaction.rs
--- a/mm2src/mm2_bitcoin/chain/src/transaction.rs	(revision e6a201f1fe27366fe6a4022f3984b121e0870a91)
+++ b/mm2src/mm2_bitcoin/chain/src/transaction.rs	(date 1683669641687)
@@ -430,6 +430,12 @@
     PosvWithNTime,
 }
 
+impl TxType {
+    fn uses_witness(&self) -> bool {
+        matches!(self, TxType::StandardWithWitness | TxType::PosvWithNTime)
+    }
+}
+
 pub fn deserialize_tx<T>(reader: &mut Reader<T>, tx_type: TxType) -> Result<Transaction, Error>
 where
     T: io::Read,
@@ -443,13 +449,13 @@
         version_group_id = reader.read()?;
     }
 
-    let n_time = if tx_type == TxType::PosWithNTime {
+    let mut n_time = if tx_type == TxType::PosWithNTime {
         Some(reader.read()?)
     } else {
         None
     };
     let mut inputs: Vec<TransactionInput> = reader.read_list_max(MAX_LIST_SIZE)?;
-    let read_witness = if inputs.is_empty() && !overwintered && tx_type == TxType::StandardWithWitness {
+    let read_witness = if inputs.is_empty() && !overwintered && tx_type.uses_witness() {
         let witness_flag: u8 = reader.read()?;
         if witness_flag != WITNESS_FLAG {
             return Err(Error::MalformedData);
@@ -461,7 +467,7 @@
         false
     };
     let outputs = reader.read_list_max(MAX_LIST_SIZE)?;
-    if outputs.is_empty() && tx_type == TxType::StandardWithWitness {
+    if outputs.is_empty() && tx_type.uses_witness() {
         return Err(Error::Custom("Transaction has no output".into()));
     }
     if read_witness {
@@ -472,6 +478,14 @@
 
     let lock_time = reader.read()?;
 
+    let mut posv = false;
+    n_time = if tx_type == TxType::PosvWithNTime {
+        posv = true;
+        Some(reader.read()?)
+    } else {
+        n_time
+    };
+
     let mut expiry_height = 0;
     let mut value_balance = 0;
     let mut shielded_spends = vec![];
@@ -521,8 +535,6 @@
         None
     };
 
-    let posv = false;
-
     Ok(Transaction {
         version,
         n_time,
@@ -547,8 +559,8 @@
 }
 
 pub fn deserialize_tx_posv<T>(reader: &mut Reader<T>, tx_type: TxType) -> Result<Transaction, Error>
-where
-    T: io::Read,
+    where
+        T: io::Read,
 {
     let version = reader.read()?;
 
@@ -639,7 +651,7 @@
         // specific use case
         let mut buffer = vec![];
         reader.read_to_end(&mut buffer)?;
-        if let Ok(t) = deserialize_tx_posv(&mut Reader::from_read(buffer.as_slice()), TxType::PosvWithNTime) {
+        if let Ok(t) = deserialize_tx(&mut Reader::from_read(buffer.as_slice()), TxType::PosvWithNTime) {
             return Ok(t);
         }
         if let Ok(t) = deserialize_tx(&mut Reader::from_read(buffer.as_slice()), TxType::StandardWithWitness) {
```